R/lint.R

```diff
@@ -140,49 +140,37 @@ reorder_lints <- function(lints) {
 #' @inherit lint_file return
 #' @examples
 #' \dontrun{
-#'   lint_package()
-#'   lint_package(
+#'   lint_dir()
+#'   lint_dir(
 #'     linters = list(semicolon_terminator_linter())
 #'     cache = TRUE,
 #'     exclusions = list("inst/doc/creating_linters.R" = 1, "inst/example/bad.R")
 #'   )
 #' }
 #' @export
-lint_package <- function(path = ".", relative_path = TRUE, ..., exclusions = NULL) {
-  path <- find_package(path)
-
-  read_settings(path)
-  on.exit(clear_settings, add = TRUE)
-
-  exclusions <- normalize_exclusions(c(exclusions, settings$exclusions), FALSE)
-
-  files <- dir(
-    path = file.path(path,
-                     c("R",
-                       "tests",
-                       "inst")
-                     ),
+lint_dir <- function(path = ".", relative_path = TRUE, ..., exclusions = NULL) {
+  files <- dir( path,
     pattern = rex::rex(".", one_of("Rr"), end),
     recursive = TRUE,
     full.names = TRUE
   )
 
   # Remove fully ignored files to avoid reading & parsing
   to_exclude <- vapply(seq_len(length(files)),
-    function(i) {
-      file <- files[i]
-      file %in% names(exclusions) && exclusions[[file]] == Inf
-     },
-    logical(1))
+                       function(i) {
```

Comment:
@arekbee I think this file will benefit from styling with styler package - https://github.com/r-lib/styler

this will make review easier.